### ISSUE TITLE
Issues with hooks from a symlinked directory, integrated via workspaces (npm)

### ISSUE THREAD

**Issue by dominikzogg** [vercel/next.js#49097]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP PREEMPT_DYNAMIC Mon May  1 00:55:28 UTC 2023
    Binaries:
      Node: 18.16.0
      npm: 9.6.5
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.3.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to the code that reproduces this issue

https://github.com/dominikzogg/reproduction-app

### To Reproduce

Use my repo with `master` branch (next: `>=13.3.2`) and its broken, use the `working-version` branch `13.3.1` and it works.

### Describe the Bug

When i upgrade next to >=13.3.2 it produces the following error:

```
../shared/hooks/use-cookie-consent.tsx (10:66) @ undefined
Unhandled Runtime Error
TypeError: dispatcher is null
```

I guess its related to: https://github.com/vercel/next.js/pull/48506

### Expected Behavior

I expect that the code runs as it did until 13.3.1

**Comment by dominikzogg** [comment]
As an information: 13.4.0 still of this issue

**Comment by dominikzogg** [comment]
Hello @ijjk

Do you have an idea what is missing there to get any feedback?

Regards Dominik

**Comment by ebo-bpi** [comment]
We seem to have this issue as well. so if you us a "yarn link" with nextjs you will get this error. 
no clue on how to solve this.

**Comment by dominikzogg** [comment]
@ebo-bpi cause i am using docker also on a local setup instead of symlinks i used docker mounts. Works but is a hack and breaks ide support cause your host system doesn't know about this references

**Comment by ebo-bpi** [comment]
@dominikzogg  yep that could be a workaround also. I "fixed" it by importing the package by file path. just point it to the output dir of the remote package. only thing here is that you need to remove this from your package.json before pushing your code. 
Again also a workaround, but it won't stop me from moving at least. 
Would be nice if this could be solved anyway.

**Comment by dominikzogg** [comment]
@ebo-bpi i would already appreciate any response from the developers.

**Comment by NickSchmitt** [comment]
Also experiencing this issue. A symlinked package via yarn link is giving me `TypeError: dispatcher is null`.
